packages/forms-web-app/src/dynamic-forms/middleware/redirect-to-unanswered-question.js

```diff
@@ -4,21 +4,19 @@ module.exports = () => (req, res, next) => {
 	const journeyResponse = res.locals.journeyResponse;
 	const journey = getJourney(journeyResponse);
 
-	const questionFieldNames = new Set(
-		journey.sections.flatMap((section) => section.questions.map((question) => question.fieldName))
-	);
+	for (const section of journey.sections) {
```

Comment:
this keeps changing, what's the actual goal of this middleware?

Reply:
could be a function on the journey to make this more resuable?

Reply:
Goal is to stop users manually changing the url to be able to get to the task list/ check your answers page until theyve answered all required questions.

As I said in a comment on one of the tickets, the check answers page is much simpler so if a user were to nav there the button to submit is on display, it isnt hidden in the journey complete section at the bottom of all sections like in other journeys.

Reply:
why can't we just hide the button then?